flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1064,6 +1076,13 @@ public void handleError(final Exception exception) {
 	 */
 	protected abstract void initialize() throws ResourceManagerException;
 
+	/**
+	 * Terminates the framework specific components.
+	 *
+	 * @throws Exception with occurs during termination.
+	 */
+	protected abstract void terminate() throws Exception;
```

Comment:
Just as a side note: As discussed with @tillrohrmann making terminate being non-blocking would be follow-up feature we might consider later on.